### PR TITLE
Use @api tag for docs

### PR DESCRIPTION
**Pull request by Jesus89**
Related to: #1764

Now there are two commands to generate docs:
- `npm run docs`  (tagged with @api)
- `npm run docs:all` (except @ignored)

Also, Travis CI deploys to gh-pages the Public docs in "docs/v4" and All docs in "docs/v4-internal".

http://cartodb.github.io/cartodb.js/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,13 +18,4 @@ notifications:
...
-
-#deploy:
-#  provider: pages
-#  skip_cleanup: true
-#  local_dir: out
-#  on:
-#    branch: jsdoc-ghpages
-#  github-token:
-#    secure: ...
+    secure: lcdjTpuv2imFSgcYWMLV4odyZ4jOTjJhyViGg9hOFzggoPGFAjplz6bPgNJsxq20ZsFDw4UN6UlZ8lPGlByIUBhf2OXs0cEJDcNSoYAV7CuTMSdZsLDb+TrWQvjtSZMBEkeVENfJjyfiJo7PCt2/cIQXSVSnhZ5Uiir+Eus5x/s=
```
Comment (IagoLast) [inline .travis.yml]: What about [this](https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings) instead using encrypted vars?

File: doc/index.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <title>CARTO.js</title>
+  <link type="text/css" rel="stylesheet" href="docs/v4/styles/jsdoc-default.css">
+</head>
+<body>
+  <h1 class="page-title">CARTO.js <span style="color:#0095dd;">v4</span></h1>
+  <ul style="margin:24px;">
+    <li><h3><a href="docs/v4">API docs</a></h3></li>
```
Comment (IagoLast) [inline doc/index.html]: should this be doc/v4?

File: .travis.yml (hunk 1 above)
Comment (Jesus89) [reply .travis.yml]: It can be done this way, but for this purpose the encrypted tokens are perfectly fine

File: doc/index.html (hunk 2 above)
Comment (Jesus89) [reply doc/index.html]: This file is the main index.html in gh-pages: https://github.com/CartoDB/cartodb.js/blob/gh-pages/index.html. It is copied with the "docs" directory. I think we could move it to another directory, may be "config/docs"

**Comment by Jesus89** [comment]
In gh-pages deploy out is moved to docs. I can generate always the "docs" directory, but this will be git-ignored.

**Comment by IagoLast** [comment]
should we change the .gitignore? :D

File: config/jsdoc/plugins/api.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
+/**
```
Comment (alonsogarciapablo) [inline config/jsdoc/plugins/api.js]: It looks like this file still contains some stuff that is specific to OpenLayers (`olx.js`)... It'd be nice to cleanup a bit so and fit it to what we need. Although it doesn't look something very straight forward.

Comment (Jesus89) [reply config/jsdoc/plugins/api.js]: Super cleanup done!

Comment (alonsogarciapablo) [reply config/jsdoc/plugins/api.js]: 🔝 !

**Pull request merged**